eodag/rest/server.py

```diff
@@ -271,42 +273,54 @@ def stac_extension_oseo(request: Request):
     return jsonable_encoder(response)
 
 
+class SearchBody(BaseModel):
+    """
+    class which describes the body of a search request
+    """
+
+    provider: Union[str, None] = None
+    collections: List[str]
```

Comment:
```suggestion
    collections: Union[List[str], str]
```
Handle single collection passed in a list or as a simple str